docs/src/examples/components/Tree/Types/TreeExampleExperiment.shorthand.tsx

```diff
@@ -0,0 +1,166 @@
+import * as React from 'react'
+import * as _ from 'lodash'
+import { Tree as StardustTree } from '@stardust-ui/react'
+
+type TreeProps = {
+  items?: Object[]
+  title?: string
+  level?: number
+  size?: number
+  id?: string
+}
+
+type TreeState = {
+  open?: boolean
+}
+
+class Tree extends React.Component<TreeProps, TreeState> {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      open: !this.props.level,
+    }
+  }
+
+  handleClick = () => {
+    if (this.props.items && this.props.items.length) {
+      this.setState(state => ({
+        open: !state.open,
+      }))
+    }
+  }
+
+  render() {
+    const { title, items, level = 0, ...rest } = this.props
+    const { open } = this.state
+
+    if (!level) {
+      return (
+        <div style={{ paddingLeft: '10px' }} role="tree">
+          {open &&
+            items.map((item, index) => (
+              <Tree
+                {...item}
+                level={level + 1}
+                aria-posinset={index}
+                id={`item-${level}-${index}`}
+              />
+            ))}
+        </div>
+      )
+    }
+
+    return (
+      <>
+        <div
```

Comment:
consider converting this to `TreeItem` custom component to improve readability of code